modulemd/v2/modulemd-module-stream-v2.c

```diff
@@ -777,6 +777,38 @@ modulemd_module_stream_v2_replace_dependencies (ModulemdModuleStreamV2 *self,
 }
 
 
+void
+modulemd_module_stream_v2_clear_dependencies (ModulemdModuleStreamV2 *self)
+{
+  g_return_if_fail (MODULEMD_IS_MODULE_STREAM_V2 (self));
+
+  g_ptr_array_set_size (self->dependencies, 0);
+}
+
+
+static gboolean
+dep_equal_wrapper (gconstpointer a, gconstpointer b)
+{
+  return modulemd_dependencies_equals ((ModulemdDependencies *)a,
+                                       (ModulemdDependencies *)b);
+}
+
+
+void
+modulemd_module_stream_v2_remove_dependencies (ModulemdModuleStreamV2 *self,
+                                               ModulemdDependencies *deps)
+{
+  guint index;
+  g_return_if_fail (MODULEMD_IS_MODULE_STREAM_V2 (self));
+
+  while (g_ptr_array_find_with_equal_func (
```

Comment:
@sgallagher, I'm not sure if this is an issue, but if there are multiple dependencies that are equal, then they will all be removed. I was expecting just the first one that was found to be removed.

For example:
```python
mmd.add_dependencies(mmd.get_dependencies()[0].copy())
assert len(mmd.get_dependencies()) == 2
mmd.remove_dependencies(mmd.get_dependencies()[0])
assert len(mmd.get_dependencies()) == 0
```